### PR TITLE
feat: use objection error types in error-handler

### LINKED ISSUE #130: Error handling is not working correctly for Postgres
### Issue

My user model requires emails to be unique. If I try to create a user with a duplicate email address I get a 500 error with the DB error as the message.

```
GeneralError: insert into "users" ("created_on", "email", "isVerified", "name", "password", "updated_on", "verifyChanges", "verifyExpires", "verifyShortToken", "verifyToken") values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) returning "id" - duplicate key value violates unique constraint "users_email_unique"
```

### Debugging

I've been trying to debug in `error-handler.js`. It looks like this error just isn't being handled. The incoming error object has all the necessary info to return a well formed Feathers error, but it's not doing anything.

### Code

I have a very simple user service that looks like this:

`users.service.ts`
```typescript
// Initializes the `user` service on path `/user`
import { ServiceAddons } from '@feathersjs/feathers';
import { Application } from '../../declarations';
import Users from './users.class';
import UserModel, { UserType } from '../../models/user.model';
import hooks from './users.hooks';

// Add this service to the service type index
declare module '../../declarations' {
  interface ServiceTypes {
    users: Users & ServiceAddons<UserType>;
  }
}

export default function (app: Application) {
  const options = {
    Model: UserModel,
    paginate: app.get('paginate'),
  };

  // Initialize our service with any options it requires
  app.use('/users', new Users(options, app));

  // Get our initialized service so that we can register hooks
  const service = app.service('users');

  service.hooks(hooks);
}
```

`users.class.ts`
```typescript
import UserModel, { UserType } from '../../models/user.model';
import Base from '../base.class';

export default class Users extends Base<UserType, UserModel> {}
```

`base.class` is very simple and is simply extending `feathers-objection.Service` with some extra typing.

### Dependencies
```javascript
{
    "feathers-objection": "^6.1.1",
    "objection": "^2.2.3",
    "pg": "^8.4.1",
    "knex": "^0.21.6",
}
```

### PR DESCRIPTION
**Pull request by robbyphillips**
This PR changes the `errorHandler` function to use the error classes created by objection.  This removes the need to parse individual error types by db type, since it's already handled by objection.

Inspired by this objection recipe: https://vincit.github.io/objection.js/recipes/error-handling.html#examples

This should close #130.

### CODE REVIEW

**Comment by dekelev** [comment]
Thanks! I'll check it out tomorrow.

בתאריך יום ה׳, 19 בנוב׳ 2020, 19:59, מאת Rob Phillips ‏<
notifications@github.com>:

**Comment by GVanderLugt** [comment]
This worked well and solved my problem. I am getting an appropriate error now and am no longer getting a 500 for my unique violation error that I described in my issue (#130). However, the error message still isn't great. It's exposing the actual DB query. I suppose this should be handled by an app level error hook. Unless we feel like writing a parser to give good human readable error messages for all of these different errors 😆 

```
insert into "users" ("created_on", "email", "isVerified", "name", "password", "updated_on", "verifyChanges", "verifyExpires", "verifyShortToken", "verifyToken") values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) returning "id" - duplicate key value violates unique constraint "users_email_unique"
```

**Comment by dekelev** [comment]
You can use the FeathersJS error hook to handle server errors before they
sent out the client and after you logged them in your system.

https://docs.feathersjs.com/api/errors.html#error-handling

בתאריך יום ו׳, 20 בנוב׳ 2020, 4:01, מאת Gerrit VanderLugt ‏<
notifications@github.com>:

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,10113 @@
 {
```
Comment (dekelev) [inline package-lock.json:1]: Why do you need to update the package-lock.json file for this PR to work?

**Review by dekelev — CHANGES_REQUESTED** [review]

**Comment by dekelev** [comment]
@robbyphillips It looks great! It seems like a breaking-change version. can you please add a migration section in the bottom of the README for migrating to v7.x? Also, the [Error handling section](https://github.com/feathersjs-ecosystem/feathers-objection#error-handling) might need an update.

File: package-lock.json (hunk 1 above)
Comment (robbyphillips) [reply package-lock.json]: I think the change is because my local npm is v7, but I didn't mean to include those changes here.  I'll revert.

**Review by dekelev — APPROVED** [review]

**Comment by dekelev** [comment]
Perfect! I'll release it tomorrow. Thank you for your contribution.

**Comment by dekelev** [comment]
v7.0.0 is released

**Comment by robbyphillips** [comment]
Thanks!

**Comment by GVanderLugt** [comment]
Thanks @robbyphillips!

**Comment by OMFG5716** [comment]
image

To add above the UniqueViolation error has the `columns` key undefined:

image

Due to columns key being undefined it's only throwing this:

image

Trying to work on this and create a separate pull request. How would you proceed? Anybody affected with similar issue?  This is going to be my first PR if I go ahead with this. The naïve solution for now is not include the columns affected just show the underlying 'sqlMessage' or better yet show the 'constraint' key in the message as a column.

**Comment by robbyphillips** [comment]
@OMFG5716  --

Objection gets the errors from the underlying sql drivers with this lib: https://github.com/Vincit/db-errors, and in the tests I just mocked the object we should be getting from that lib.

So first step would be to see if we can figure out why we're not getting what's expected, possibly because that `db-errors` lib returns a different shape for mysql or whatever.  If that's true, then the solution is to add a different conditional to check for `error.columns` before making the error message.

**Comment by robbyphillips** [comment]
The error parser for mysql definitely returns a different shape:  https://github.com/Vincit/db-errors/blob/6ed1975c135484e127643250e1e64b2bd8fc35a8/lib/parsers/mysql/DBError/ConstraintViolationError/UniqueViolationError/parser.js#L20

My test was based on the error constructor defined here: https://github.com/Vincit/db-errors/blob/master/lib/errors/UniqueViolationError.js, but it was a bad assumption on my part that all those keys would actually exist for every parser.

**Comment by dekelev** [comment]
@OMFG5716 Issue is fixed in v7.1.2

**Pull request merged**